dev/io.openliberty.microprofile.openapi.internal.common/src/io/openliberty/microprofile/openapi/internal/common/OpenAPIEndpointManager.java

```diff
@@ -0,0 +1,181 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component(configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true, configurationPid = "io.openliberty.microprofile.openapi")
+public class OpenAPIEndpointManager {
+
+    private static final TraceComponent tc = Tr.register(OpenAPIEndpointManager.class);
+
+    private static final String OPEN_API_DOC_VAR_NAME = "openAPIDOCPATH";
+    private static final String OPEN_API_DOC_ENDPOINT_CONFIG_NAME = "docPath";
+    private static final String OPEN_API_DOC_ENDPOINT_PATH = "/openapi";
+
+    private static final String OPEN_API_UI_VAR_NAME = "openAPIUIPATH";
+    private static final String OPEN_API_UI_CONFIG_NAME = "uiPath";
+    private static final String OPEN_API_UI_PATH = "/ui";
+
+    private static final Pattern PATH_PATTERN = Pattern.compile("^(/[\\w./_-]*)?$");
+    private static final Map<String, String> CONFLICTING_PATHS_MAP = Stream.of(
+                                                                               new AbstractMap.SimpleEntry<>("/ibm/api", "OpenAPI"),
+                                                                               new AbstractMap.SimpleEntry<>("/health", "mpHealth"),
+                                                                               new AbstractMap.SimpleEntry<>("/metrics", "mpMetrics")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    public enum EndpointId {
+        UI,
+        DOCUMENT
+    };
+
+    private String uiPath = null;
+    private String docPath = null;
+
+    private WABConfigManager uiWabConfigManager;
+    private WABConfigManager docWabConfigManager;
+
+    @Activate
+    @Modified
+    protected void activate(BundleContext context, Map<String, Object> properties) {
+        docPath = OPEN_API_DOC_ENDPOINT_PATH;
+        uiPath = docPath + OPEN_API_UI_PATH;
+
+        if (ProductInfo.getBetaEdition()) {
+            //check for system property `open_api_path_enabled` as additional guide - getBoolean returns `true` if the value exists and is set to `true`, if the value is `false`
+            if (Boolean.getBoolean("open_api_path_enabled")) {
+                resolvePathFromProperties(properties);
+            }
+        }
+        //If One of the WAB Config Managers exists then we are modifying the configuration therefore we need to unregister the existing WABs
+        if(uiWabConfigManager != null){
+            deactivate();
+        }
+        uiWabConfigManager = new WABConfigManager(context, OPEN_API_UI_VAR_NAME, uiPath, "OpenAPI UI");
+        docWabConfigManager = new WABConfigManager(context, OPEN_API_DOC_VAR_NAME, docPath, "OpenAPI Doc");
+        uiWabConfigManager.register();
+        docWabConfigManager.register();
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        uiWabConfigManager.unregister();
+        docWabConfigManager.unregister();
+    }
+
+    private void resolvePathFromProperties(Map<String, Object> properties) {
```

Comment:
This isn't really resolving. Suggest renaming to `getPathsFromProperties` and adding javadoc.